File: Challenges/ReverseString.py
class Solution:
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        return s[::-1]

class Solution:
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        s.reverse()

"""
Write a function that reverses a string. The input string is given as an array of characters char[].

Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.

You may assume all the characters consist of printable ascii characters.
"""
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        for i in range(len(s)//2):
            temp = s[i]
            s[i] = s[len(s)-i-1]
            s[len(s)-i-1] = temp
            
            
            

File: Challenges/reverseString.java
class Solution {
    public void reverseString(char[] s) {
        int stringLength = s.length;
        for (int i=0; i<stringLength/2; i++) {
            char temp = s[i];
            s[i] = s[stringLength-i-1];
            s[stringLength-i-1] = temp;
        }
    }
}
